ui/imports/shared/views/AssetsViewNew.qml

```diff
@@ -0,0 +1,331 @@
+
+            Behavior on Layout.preferredHeight {
+                NumberAnimation { duration: 200; easing.type: Easing.InOutQuad }
+            }
+
+            Behavior on opacity {
+                NumberAnimation { duration: 200; easing.type: Easing.InOutQuad }
+            }
+
+            StatusDialogDivider { Layout.fillWidth: true }
+
+            RowLayout {
+                Layout.fillWidth: true
+                Layout.fillHeight: false
+
+                spacing: Style.current.halfPadding
+
+                StatusBaseText {
+                    color: Theme.palette.baseColor1
+                    font.pixelSize: Style.current.additionalTextSize
+                    text: qsTr("Sort by:")
+                }
+
+                SortOrderComboBox {
+                    id: sortOrderComboBox
+
+                    objectName: "cmbTokenOrder"
+                    hasCustomOrderDefined: root.customOrderAvailable
+
+                    model: [
+                        { value: SortOrderComboBox.TokenOrderCurrencyBalance,
+                            text: qsTr("Asset balance value"), icon: "", sortRoleName: "marketBalance" },
+                        { value: SortOrderComboBox.TokenOrderBalance,
+                            text: qsTr("Asset balance"), icon: "", sortRoleName: "balance" },
+                        { value: SortOrderComboBox.TokenOrderCurrencyPrice,
+                            text: qsTr("Asset value"), icon: "", sortRoleName: "marketPrice" },
+                        { value: SortOrderComboBox.TokenOrder1DChange,
+                            text: qsTr("1d change: balance value"), icon: "", sortRoleName: "change1DayFiat" },
+                        { value: SortOrderComboBox.TokenOrderAlpha,
+                            text: qsTr("Asset name"), icon: "", sortRoleName: "name" },
+                        { value: SortOrderComboBox.TokenOrderCustom,
+                            text: qsTr("Custom order"), icon: "", sortRoleName: "position" },
+                        { value: SortOrderComboBox.TokenOrderNone,
+                            text: "---", icon: "", sortRoleName: "" }, // separator
+                        { value: SortOrderComboBox.TokenOrderCreateCustom,
+                            text: hasCustomOrderDefined ? qsTr("Edit custom order →") : qsTr("Create custom order →"),
+                            icon: "", sortRoleName: "" }
+                    ]
+                    onCreateOrEditRequested: {
+                        root.manageTokensRequested()
+                    }
+                }
+            }
+
+            StatusDialogDivider { Layout.fillWidth: true }
+        }
+
+        DelegateModel {
+            id: regularModel
+
+            model: sfpm
+
+            delegate: TokenDelegateNew {
+                width: ListView.view.width
+
+                name: model.name
+                icon: model.icon
+                balance: `${model.balanceText} ${model.symbol}`
```

Comment:
Here I intentionally take `balanceText` as a localized and formatted text to cut-off dependence on the backend. In original `AssetsView` it was done via `currencyStore.formatCurrencyAmount`, because different tokens may have different formatting rules, that info comes from the backend.